griptape/config/azure_openai_structure_config.py

```diff
@@ -0,0 +1,97 @@
+from typing import Callable, Optional
+from attrs import Factory, define, field
+
+from griptape.config import StructureConfig
+from griptape.drivers import (
+    LocalVectorStoreDriver,
+    AzureOpenAiChatPromptDriver,
+    AzureOpenAiEmbeddingDriver,
+    AzureOpenAiImageGenerationDriver,
+    AzureOpenAiVisionImageQueryDriver,
+)
+
+
+@define
+class AzureOpenAiStructureConfig(StructureConfig):
+    """Azure OpenAI Structure Configuration.
+
+    Attributes:
+        azure_endpoint: The endpoint for the Azure OpenAI instance.
+        azure_ad_token: An optional Azure Active Directory token.
+        azure_ad_token_provider: An optional Azure Active Directory token provider.
+        api_key: An optional Azure API key.
+        prompt_driver: An Azure OpenAI Chat Prompt Driver.
+        image_generation_driver: An Azure OpenAI Image Generation Driver.
+        image_query_driver: An Azure OpenAI Vision Image Query Driver.
+        embedding_driver: An Azure OpenAI Embedding Driver.
+        vector_store_driver: A Local Vector Store Driver.
+    """
+
+    azure_endpoint: str = field(kw_only=True, metadata={"serializable": True})
+    azure_ad_token: Optional[str] = field(kw_only=True, default=None, metadata={"serializable": False})
+    azure_ad_token_provider: Optional[Callable[[], str]] = field(
+        kw_only=True, default=None, metadata={"serializable": False}
+    )
+    api_key: str = field(kw_only=True, default=None, metadata={"serializable": False})
+    prompt_driver: AzureOpenAiChatPromptDriver = field(
```

Comment:
All of the Driver type hints should use the base class so that they can be overridden with non-azure drivers.